code/modules/mining/mine_turfs.dm

```diff
@@ -28,26 +30,108 @@
 	var/excav_overlay = ""
 	var/obj/item/weapon/last_find
 	var/datum/artifact_find/artifact_find
+	var/ignore_mapgen
 
 	has_resources = 1
 
-/turf/simulated/mineral/New()
-	spawn(0)
-		MineralSpread()
-	spawn(2)
-		updateMineralOverlays(1)
+/turf/simulated/mineral/ignore_mapgen
+	ignore_mapgen = 1
 
-/turf/simulated/mineral/proc/updateMineralOverlays(var/update_neighbors)
-	var/list/step_overlays = list("s" = NORTH, "n" = SOUTH, "w" = EAST, "e" = WEST)
-	for(var/direction in step_overlays)
-		var/turf/turf_to_check = get_step(src,step_overlays[direction])
-		if(update_neighbors && istype(turf_to_check,/turf/simulated/floor/asteroid))
-			var/turf/simulated/floor/asteroid/T = turf_to_check
-			T.updateMineralOverlays()
-		else if(istype(turf_to_check,/turf/space) || istype(turf_to_check,/turf/simulated/floor))
-			turf_to_check.overlays += image('icons/turf/walls.dmi', "rock_side_[direction]")
+/turf/simulated/mineral/floor
+	name = "sand"
+	icon = 'icons/turf/flooring/asteroid.dmi'
+	icon_state = "asteroid"
+	density = 0
+	opacity = 0
+
+/turf/simulated/mineral/floor/ignore_mapgen
+	ignore_mapgen = 1
+
+/turf/simulated/mineral/proc/make_floor()
+	if(!density && !opacity)
+		return
+	density = 0
+	opacity = 0
+	update_icon()
+	reconsider_lights()
+
+/turf/simulated/mineral/proc/make_wall()
+	if(density && opacity)
+		return
+	density = 1
+	opacity = 1
+	update_icon()
+	reconsider_lights()
+
+/turf/simulated/mineral/Entered(atom/movable/M as mob|obj)
+	. = ..()
+	if(istype(M,/mob/living/silicon/robot))
+		var/mob/living/silicon/robot/R = M
+		if(R.module)
+			if(istype(R.module_state_1,/obj/item/weapon/storage/bag/ore))
+				attackby(R.module_state_1,R)
+			else if(istype(R.module_state_2,/obj/item/weapon/storage/bag/ore))
```

Comment:
This is all existing code that has been collapsed, I was not intending to volunteer for a total refactor of this bollocks.